google_compute_engine/metadata_watcher.py

```diff
@@ -178,6 +180,10 @@ def _HandleMetadataUpdate(
           exception = e
           self.logger.error('GET request error retrieving metadata. %s.', e)
 
+      if not retry:
+        # If it didn't returned up to this point, and exception occured.
+        break
```

Comment:
I don't think the following is possible in the current code, but it's conceivable that `exception` will eventually be set, in which case the continue will be triggered before this break. To prevent that, I think you should move the if statement to the first line of the except (and you can drop the comment).

Reply:
I see the advantages of having it as I wrote as the (desirable) error message `GET request ...` will be displayed on failure.

I don't think the `continue` can be reached, as when `retry == False`, the `exception` variable will be considered dead code. I explain: The `continue` on this function will never be triggered as no exception class defined on `except` clause is of type `NoneType`. Loop will reach a break after that, leaving no possibility, in this case, for continuing the `while` loop.

Do you agree in keeping it like I proposed or do you want me to change?

Reply:
I believe that you are correct that in the current code the continue cannot be hit, but I also think it is possible that behavior may change in the future. I do agree that the logging is desirable - great catch. How about we update the logic to this:

```
if not isinstance(e, type(exception)):
  exception = e
  self.logger.error('GET request error retrieving metadata. %s.', e)
if retry:
  continue
else:
  break
```